sdks/java/core/src/test/java/org/apache/beam/sdk/metrics/MetricsTest.java

```diff
@@ -287,12 +288,19 @@ public void testBoundedSourceMetrics() {
 
       assertThat(
           metrics.getCounters(),
-          hasItem(
-              attemptedMetricsResult(
-                  ELEMENTS_READ.getNamespace(),
-                  ELEMENTS_READ.getName(),
-                  "Read(BoundedCountingSource)",
-                  1000L)));
+          anyOf(
+              hasItem(
+                  attemptedMetricsResult(
+                      ELEMENTS_READ.getNamespace(),
+                      ELEMENTS_READ.getName(),
+                      "Read(BoundedCountingSource)",
```

Comment:
Why the divergence here? Is it portable vs. non-portable?

Reply:
Yes.